linesearch.go

```diff
@@ -63,46 +60,34 @@ func (ls *LinesearchMethod) Iterate(loc *Location) (Operation, error) {
 		// Init was not called. What to do? What about ls's internal state?
```

Comment:
Calling Iterate() without calling Init() could indeed panic, but in this case I was considering the situation when Iterate() returned with an error and the caller is trying to resume the optimization after doing something about the error without calling Init() (because that is what we discussed previously). `Local` precludes this use case and always calls Init(), but I still operate under the assumption that Local is a kind of convenience wrapper around the iteration loop and someone might want more flexibility and use its own loop.

Calling Init() with the location at the last MajorIteration seems to be the best universal solution but I had the impression that you did not want to enforce it. With the previous big PR, this one and with the changes related to removing Finished() from Linesearcher that I have in my mind I could not sort out my thoughts on this.